pycaret/loggers/dagshub_logger.py

```diff
@@ -0,0 +1,94 @@
+import os
+from pathlib import Path
+
+from pycaret.loggers.mlflow_logger import MlflowLogger
+
+try:
+    import dagshub
+    import mlflow
+    from dagshub.upload import Repo
+except ImportError:
+    dagshub = None
+
+
+class DagshubLogger(MlflowLogger):
+    def __init__(self, remote=None) -> None:
+        super().__init__()
+        if dagshub is None:
+            raise ImportError(
+                "DagshubLogger requires dagshub. Install using `pip install dagshub`"
+            )
+        # check token exist or not:
+        token = dagshub.auth.get_token()
+        os.environ["MLFLOW_TRACKING_USERNAME"] = token
```

Comment:
is it possible to avoid modifying env vars?

Reply:
We could try caching or writing to temporary storage, but I'd assume that'll make the code needlessly convoluted

Reply:
MLflow uses env vars to log experiments to remote servers. It's the default behavior defined by MLflow. I think we shouldn't reinvent the wheel and stick with the flow MLflow users are used to.